fedimint-client/src/lib.rs

```diff
@@ -2245,15 +2251,37 @@ impl ClientBuilder {
         Ok(client)
     }
 
-    pub async fn open(self, root_secret: DerivableSecret) -> anyhow::Result<ClientHandle> {
+    pub async fn open(self, pre_root_secret: DerivableSecret) -> anyhow::Result<ClientHandle> {
         let Some(config) = Client::get_config_from_db(&self.db_no_decoders).await else {
             bail!("Client database not initialized")
         };
 
+        if let Some(secret_hash) = self
+            .db_no_decoders()
+            .begin_transaction_nc()
+            .await
+            .get_value(&ClientPreRootSecretHashKey)
+            .await
+        {
+            if pre_root_secret.derive_pre_root_secret_hash() != secret_hash {
+                bail!("Secret hash does not match. Incorrect secret")
+            }
+
+            let mut dbtx = self.db_no_decoders.begin_transaction().await;
+            dbtx.insert_entry(
+                &ClientPreRootSecretHashKey,
+                &pre_root_secret.derive_pre_root_secret_hash(),
+            )
+            .await;
+            dbtx.commit_tx().await;
```

Comment:
This should be in an else branch, shouldn't it? If the hash is in the DB and matches we don't need to write it again.

Reply:
Oops.